sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java

```diff
@@ -82,10 +80,14 @@ public synchronized Integer addEmsDeployment(String sessionId, EmsDeploymentDefi
 
         AtomicInteger failedDeploymentIdentification = new AtomicInteger();
         // For supplied node ...
-        emsDeploymentDefinition.getNodeNames()
-                               .forEach(node -> addEmsDeploymentForNode(node,
-                                                                        emsDeploymentDefinition.getAuthorizationBearer(),
-                                                                        emsDeploymentDefinition.isPrivateIP()));
+        emsDeploymentDefinition.getNodeNames().forEach(nodeName -> {
+            Deployment deployment = repositoryService.getDeployment(nodeName);
+            addEmsDeploymentForNode(deployment,
+                                    emsDeploymentDefinition.getAuthorizationBearer(),
+                                    emsDeploymentDefinition.isPrivateIP());
+            repositoryService.saveDeployment(deployment);
+            repositoryService.flush();
```

Comment:
Is there a reason of why you flush all entities here ? `repositoryService.saveDeployment(deployment);` already saves and flush deployments entities.

If you do need to flush all entities, I think you can keep just the one below, after the loop and not flush at each iteration.

Reply:
You're right. I should have missed to delete that one.